src/rust.rs

```diff
@@ -965,3 +955,79 @@ pub static SIMPLE_GLOBAL_CLASS: JSClass = JSClass {
     cOps: &SIMPLE_GLOBAL_CLASS_OPS as *const JSClassOps,
     reserved: [0 as *mut _; 3]
 };
+
+#[inline]
+unsafe fn get_object_group(obj: *mut JSObject) -> *mut ObjectGroup {
+    assert!(!obj.is_null());
+    (*(obj as *mut Object)).group
```

Comment:
I think this would be more readable if you added a separate `let` binding for `obj as *mut Object`.